### DISCUSSION TITLE
Vault as a external pki provider

### DISCUSSION THREAD

**Discussion by ysh78NJ** [linkerd/linkerd2#7867] [Q&A] [answered]
Was just wondering if there was any documentation on how to use Vault as a PKI provider for linkerd's mTLS?   I saw an announcement for that but can't seem to locate anything further.

**Comment by olix0r** [comment, marked as answer]
@ysh78NJ you can use [cert-manager](https://linkerd.io/2.11/tasks/automatically-rotating-control-plane-tls-credentials/) to manage linkerd's issuer credentials. cert-manager can be [configured to use Vault](https://cert-manager.io/docs/configuration/vault/).

**Comment by d3xma** [comment]
@ysh78NJ I was just trying to set this up, but ran into [this issue](https://github.com/linkerd/linkerd2/issues/10391).
Any ideas on how to resolve this?

**Comment by channyein87** [comment]
Linkerd has created a demo for using enterpise pki like venafi and vault.

https://www.youtube.com/watch?v=9InXEc1TULU
https://github.com/BuoyantIO/service-mesh-academy/tree/main/l5d-enterprise-pki
